src/index.ts

```diff
@@ -90,104 +78,100 @@ export interface IImageName {
-      handler: 'bootstrap',
-      environment: props.environment,
+      code: lambda.Code.fromAsset(path.join(__dirname, '../lambda')),
+      runtime: lambda.Runtime.PYTHON_3_11,
+      handler: 'index.on_event',
+      environment: Object.assign({
+        // NOTICE: Change the default credentials store location.
+        // https://github.com/google/go-containerregistry/blob/8dadbe76ff8c20d0e509406f04b7eade43baa6c1/pkg/authn/README.md?plain=1#L45
+        DOCKER_CONFIG: '/tmp/.docker',
+      }, props.environment),
       lambdaPurpose: 'Custom::CDKECRDeployment',
       timeout: Duration.minutes(15),
       role: props.role,
-      memorySize: memoryLimit,
       vpc: props.vpc,
+      memorySize: memoryLimit,
       vpcSubnets: props.vpcSubnets,
       securityGroups: props.securityGroups,
+      initialPolicy: [
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            'ecr:GetAuthorizationToken',
+            'ecr:BatchCheckLayerAvailability',
+            'ecr:GetDownloadUrlForLayer',
+            'ecr:GetRepositoryPolicy',
+            'ecr:DescribeRepositories',
+            'ecr:ListImages',
+            'ecr:DescribeImages',
+            'ecr:BatchGetImage',
+            'ecr:ListTagsForResource',
+            'ecr:DescribeImageScanFindings',
+            'ecr:InitiateLayerUpload',
+            'ecr:UploadLayerPart',
+            'ecr:CompleteLayerUpload',
+            'ecr:PutImage',
+          ],
+          resources: ['*'],
+        }),
+      ],
+      layers: [
+        CraneLayer.getInstance(scope),
+      ],
     });
 
-    const handlerRole = this.handler.role;
-    if (!handlerRole) { throw new Error('lambda.SingletonFunction should have created a Role'); }
-
-    handlerRole.addToPrincipalPolicy(
-      new iam.PolicyStatement({
-        effect: iam.Effect.ALLOW,
-        actions: [
-          'ecr:GetAuthorizationToken',
-          'ecr:BatchCheckLayerAvailability',
-          'ecr:GetDownloadUrlForLayer',
-          'ecr:GetRepositoryPolicy',
-          'ecr:DescribeRepositories',
-          'ecr:ListImages',
-          'ecr:DescribeImages',
-          'ecr:BatchGetImage',
-          'ecr:ListTagsForResource',
-          'ecr:DescribeImageScanFindings',
-          'ecr:InitiateLayerUpload',
-          'ecr:UploadLayerPart',
-          'ecr:CompleteLayerUpload',
-          'ecr:PutImage',
-        ],
-        resources: ['*'],
-      }));
-    handlerRole.addToPrincipalPolicy(new iam.PolicyStatement({
-      effect: iam.Effect.ALLOW,
-      actions: [
-        's3:GetObject',
-      ],
-      resources: ['*'],
-    }));
+    const provider = new cr.Provider(this, 'Provider', {
+      onEventHandler: this.handler,
+    });
 
     new CustomResource(this, 'CustomResource', {
-      serviceToken: this.handler.functionArn,
+      serviceToken: provider.serviceToken,
       resourceType: 'Custom::CDKBucketDeployment',
       properties: {
+        Time: Date.now().toString(),
```

Comment:
This behavior can/should probably be triggered more intelligently. For example, by examining whether or not a tag is supplied, or adding a `triggerEveryDeployment?: boolean`/`mutableSourceUri?: boolean` property.

And in any case, functionality changes should not be part of the PR whose sole purpose it is to make the package behave more reliably.